week-1/2-mandatory/4-tax.js

```diff
@@ -1,29 +1,18 @@
-/*
-  SALES TAX
-  =========
-  A business requires a program that calculates how much sales tax to charge
-  Sales tax is 20% of the price of the product
-*/
-
-function calculateSalesTax() {}
-
-/*
-  CURRENCY FORMATTING
-  ===================
-  The business has informed you that prices must have 2 decimal places
-  They must also start with the currency symbol
-  Write a function that transforms numbers into the format £0.00
-
-  Remember that the prices must include the sales tax (hint: you already wrote a function for this!)
-*/
-
-function formatCurrency() {}
+function sTax(Price) {
+  let salesTax = 0.2 * Price;
+  return salesTax;
+}
 
-/* ======= TESTS - DO NOT MODIFY ===== 
-There are some Tests in this file that will help you work out if your code is working.
+function calculateSalesTax(Price) {
+  let tax = sTax(Price);
+  return tax + Price;
+}
 
-To run these tests type `node 4-tax.js` into your terminal
-*/
+function formatCurrency(Price) {
+  let tax = sTax(Price);
+  lastPrice = Price + tax;
+  return "£" + lastPrice.toFixed(2);
+}
 
 function test(test_name, expr) {
   let status;
```

Comment:
This is well done. 

A couple of notes FYI. 

Nothing you can do but - one of the tests is badly written in my opinion.
the test for `formatCurrency` demands that the "calculation of price" should also be included. 
This would be better named `getFullPriceAsFormattedCurrency` as it does much more than simply "formatCurrency".

I would keep actually keep the function as `formatCurrency` which just accepts a value and returns the same value as a currency. 
This allows us to reuse it again and again anytime we need to print a value as currency.
Then to print the fullPrice in currency you would run 

`formatCurrency( calculateFullPrice(15) )`

With that in mind we can simplify this program for the reader.

```
function calculatePercentage(percent, value) {
  return (percent / 100) * value
}

function calculateSalesTax(price) {
  return calculatePercentage(20, price)
}

function calculateFullPrice(price) {
  return calculateSalesTax(price) + price
}

function formatCurrency(value) {
  return `£${value.toFixed(2)}`
}
```

Again nothing you did wrong here - just some extra info